File: modules/core/misc/java/test/DMatchTest.java
package org.opencv.test.core;

import org.opencv.core.DMatch;

import junit.framework.TestCase;

public class DMatchTest extends TestCase {
    public void testDMatch() {
        new DMatch();
    }

    public void testDMatchIntIntFloat() {
        DMatch dm1 = new DMatch(1, 4, 4.0f);

        assertEquals(1, dm1.queryIdx);
        assertEquals(4, dm1.trainIdx);
        assertEquals(4.0f, dm1.distance);
    }

    public void testDMatchIntIntIntFloat() {
        DMatch dm2 = new DMatch(2, 6, -1, 8.0f);

        assertEquals(2, dm2.queryIdx);
        assertEquals(6, dm2.trainIdx);
        assertEquals(-1, dm2.imgIdx);
        assertEquals(8.0f, dm2.distance);
    }

    public void testLessThan() {
        DMatch dm1 = new DMatch(1, 4, 4.0f);
        DMatch dm2 = new DMatch(2, 6, -1, 8.0f);
        assertTrue(dm1.lessThan(dm2));
    }

    public void testToString() {
        DMatch dm2 = new DMatch(2, 6, -1, 8.0f);

        String actual = dm2.toString();

        String expected = "DMatch [queryIdx=2, trainIdx=6, imgIdx=-1, distance=8.0]";
        assertEquals(expected, actual);
    }

}

File: modules/core/misc/objc/test/DMatchTest.swift
//
//  DMatchTest.swift
//
//  Created by Giles Payne on 2020/01/31.
//

import XCTest
import OpenCV

class DMatchTest: OpenCVTestCase {

    func testDMatchIntIntFloat() {
        let dm1 = DMatch(queryIdx: 1, trainIdx: 4, distance: 4.0)

        XCTAssertEqual(1, dm1.queryIdx)
        XCTAssertEqual(4, dm1.trainIdx)
        XCTAssertEqual(4.0, dm1.distance)
    }

    func testDMatchIntIntIntFloat() {
        let dm2 = DMatch(queryIdx: 2, trainIdx: 6, imgIdx: -1, distance: 8.0)

        XCTAssertEqual(2, dm2.queryIdx)
        XCTAssertEqual(6, dm2.trainIdx)
        XCTAssertEqual(-1, dm2.imgIdx)
        XCTAssertEqual(8.0, dm2.distance)
    }

    func testLessThan() {
        let dm1 = DMatch(queryIdx: 1, trainIdx: 4, distance: 4.0)
        let dm2 = DMatch(queryIdx: 2, trainIdx: 6, imgIdx: -1, distance: 8.0)
        XCTAssert(dm1.lessThan(dm2))
    }

    func testToString() {
        let dm2 = DMatch(queryIdx: 2, trainIdx: 6, imgIdx: -1, distance: 8.0)

        let actual = "\(dm2)"

        let expected = "DMatch { queryIdx: 2, trainIdx: 6, imgIdx: -1, distance: 8.000000}"
        XCTAssertEqual(expected, actual)
    }

}
